group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -367,6 +389,55 @@ public int partitionFor(
         return Utils.abs(groupId.hashCode()) % numPartitions;
     }
 
+    /**
+     * Validates the request.
+     *
+     * @param request The request to validate.
+     * @param apiVersion The version of ConsumerGroupHeartbeat RPC
+     * @throws InvalidRequestException if the request is not valid.
+     * @throws UnsupportedAssignorException if the assignor is not supported.
+     */
+    private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
```

Comment:
This is a copy of the method that was in the manager. The only difference is `!consumerGroupAssignors.contains(request.serverAssignor())`.